include/oneapi/dpl/internal/dynamic_selection_impl/scheduler_defs.h

```diff
@@ -0,0 +1,38 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_SCHEDULER_DEFS_H
+#define _ONEDPL_SCHEDULER_DEFS_H
+
+namespace oneapi {
+namespace dpl{
+namespace experimental{
+
+  template<typename NativeContext>
+  struct basic_execution_resource_t {
```

Comment:
A name suggestion for future consideration: `trivial_execution_resource`; similarly for other types. Not sure if it makes a big difference, but to me "trivial" better reflects the idea that these types really do nothing.